footbot/main.py

```diff
@@ -262,13 +262,23 @@ def update_predictions_route():
     return "predictions updated"
 
 
-@app.route("/optimise_team/<entry>")
-def optimise_team_route(entry):
+@app.route("/optimise_team/<entry>", methods=["GET", "POST"])
+def optimise_team_route(entry, optimise_entry=team_selector.optimise_entry):
 
     bootstrap_data = requests.get(
         "https://fantasy.premierleague.com/api/bootstrap-static/"
     ).json()
 
+    login = password = None
+    try:
+        data = request.json
+        login = data["login"]
+        password = data["password"]
+    except TypeError:
+        pass
+    except KeyError:
+        return "Data must contain 'login' and 'password'", 400
```

Comment:
Is returning `(msg, status_code)` tuple flask magic?

Reply:
yep- https://flask.palletsprojects.com/en/1.1.x/quickstart/#about-responses